tests/test_ess_csc.py

```diff
@@ -458,3 +458,70 @@ def topic_callback(self, data: salobj.BaseMsgType, attr_name: str) -> None:
             else:
                 sensor_data[data.timestamp] = data
         self.data_event.set()
+
+    async def test_rpi_data_client_loses_single_connecton(self) -> None:
+        """The CSC should fault when the RPiDataClient loses its connection
+        to the server and the RpiDataClient should reconnect when the CSC is
+        set to ENABLED again.
+        """
+        # Start the MockServer for manual control.
+        await self.start_mock_server()
+        async with self.make_csc(
+            initial_state=salobj.State.STANDBY,
+            config_dir=TEST_CONFIG_DIR,
+            simulation_mode=0,
+            override="test_one_temp_sensor.yaml",
+        ):
+            await self.assert_next_summary_state(salobj.State.STANDBY)
+
+            await salobj.set_summary_state(
+                remote=self.remote, state=salobj.State.DISABLED
+            )
+            await self.assert_next_summary_state(salobj.State.DISABLED)
+
+            await salobj.set_summary_state(
+                remote=self.remote, state=salobj.State.ENABLED
+            )
+            await self.assert_next_summary_state(salobj.State.ENABLED)
+            await self.assert_next_sample(topic=self.remote.evt_errorCode, errorCode=0)
+            assert len(self.csc.data_clients) == 1
+
+            await salobj.set_summary_state(
+                remote=self.remote, state=salobj.State.DISABLED
+            )
+            await self.assert_next_summary_state(salobj.State.DISABLED)
+
+            await salobj.set_summary_state(
+                remote=self.remote, state=salobj.State.STANDBY
+            )
+            await self.assert_next_summary_state(salobj.State.STANDBY)
+
+            await salobj.set_summary_state(
+                remote=self.remote, state=salobj.State.DISABLED
+            )
+            await self.assert_next_summary_state(salobj.State.DISABLED)
+
+            await salobj.set_summary_state(
+                remote=self.remote, state=salobj.State.ENABLED
+            )
+            await self.assert_next_summary_state(salobj.State.ENABLED)
+
+        # Stop the MockServer to clean up after ourselves.
+        await self.stop_mock_server()
+
+    async def start_mock_server(self) -> None:
+        self.mock_server = common.SocketServer(
+            name="MockServer",
+            host=tcpip.LOCAL_HOST,
+            port=5000,
+            simulation_mode=1,
+        )
+        mock_command_handler = common.MockCommandHandler(
+            callback=self.mock_server.write,
+            simulation_mode=1,
+        )
+        self.mock_server.set_command_handler(mock_command_handler)
+        await asyncio.wait_for(self.mock_server.start_task, timeout=STD_TIMEOUT)
+
+    async def stop_mock_server(self) -> None:
+        await self.mock_server.close()
```

Comment:
No, I am merely trying to disconnect and connect again. The reason I am starting the MockServer manually is because in simulation mode the RpiDataClient starts and stops the MockServer and it needs to keep on running in order to trigger the situation that I was trying to solve.

Reply:
I think I understand. Your test is to send the CSC to standby and re-enable it to test the bug you discovered and fixed. That is a great idea, but it does not seem to match the name of the test method, nor the doc string:

The CSC should fault when the RPiDataClient loses its connection to the server and the RpiDataClient should reconnect when the CSC is set to ENABLED again.

Please rename the method and update the doc string.

Please also consider adding a test that *does* match the doc string (if you don't already have one). I think it would be very useful. And running the mock server separately (as you do here) seems the obvious way to write the test. I hope it would be a fairly minor (but important) variant of the test you have here. The existing name and doc string would be perfect for it.

Reply:
Very good suggestion to add that second test as well, thanks. Done!